### PR TITLE
[circleci] adapt circle.yml to osx environment

### PR DESCRIPTION
**Pull request by springmeyer**
This PR modifies the circle.yml to be able to be run on the circle OS X environment. It builds off of #2958.

The overall motivation is to be able to run all tests without segfaults (that are common on linux due to uncontrolled factors: bugs in headless-gl or mesa?).

Key changes:

  - Installs node with nvm because circleci OS X machines do not support:

```
 machine:
 -  node:
 -    version: '4'
```

  - Provides new environment options: `XCODE_SCHEME` and `XCODE_WORKSPACE` to trigger no-op in circleci configuration setup.

Background: The circleci OS X support assumes you are developing iOS applications with an xcodeproject (https://circleci.com/docs/ios-builds-on-os-x/). So this project will fail by default when an xcode project is not found. This PR fixes this problem by telling circleci to not look for xcode files.

More notes:

   - Speed is slightly slower than the previous build that passed on master
     - Build was: https://circleci.com/gh/mapbox/mapbox-gl-js/4346
     - Total time: 13:19 master/linux, 15:14 PR/osx
     - Install deps from cache: 1:08 master/linux, 0:38 PR/osx
     - Save cache: 0:02 master/linux, 1:06 PR/osx
     - Run build/tests: 11:04 master/linux, 12:23 PR/osx

### CODE REVIEW

**Comment by lucaswoj** [comment]
Looks good to me. 🚢

File: ci.sh [hunk 1]
```diff
@@ -1,8 +1,36 @@
...
+}
+
+
+install_node 4
+
+if [[ "$(git symbolic-ref --short -q HEAD)" == "master" ]]; then
+    npm update
+else
+    npm install
+fi
```
Comment (lucaswoj) [inline ci.sh]: Are there any consequences of moving this from the `dependencies` step to the `test` step?  Or is that just presentational?

Comment (lucaswoj) [inline ci.sh]: My reading of the CircleCI docs is that these sections are entirely presentational https://circleci.com/docs/configuration/

Comment (springmeyer) [inline ci.sh]: Good catch: there is a consequence: caching. It looks like the cache is restored before the `Dependencies` section and saved after and before the `test` section. So to benefit from the `node_modules` cache we'd need things to be installed in the `Dependencies`. Hrm...

Comment (lucaswoj) [inline ci.sh]: I’m open to using the pre-installed node v6.

That simplifies `ci.sh` and enables dependency caching.

If upstream image changes become a problem in the future we can look into solutions then.

**Comment by lucaswoj** [comment]
This looks good. :ship: on green.

**Comment by lucaswoj** [comment]
# Next Steps

 - [ ] merge this PR
 - [ ] un-ignore as many test-suite cases as possible
 - [ ] fix any PRs that began failing during the CI debugging process

**Pull request merged**